engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -644,6 +647,37 @@ public void startApplication() throws YarnException, IOException
     }
   }
 
+  /**
+   * Process the SSLConfig if present in the DAG's attributes
+   *
+   * @param sslConfig  SSLConfig object derived from SSL_CONFIG attribute
+   * @param fs    HDFS file system object
+   * @param appPath    application path for the current application
+   * @param localResources  Local resources to modify
+   * @throws IOException
+   */
+  private void processSSLConfig(SSLConfig sslConfig, FileSystem fs, Path appPath, Map<String, LocalResource> localResources) throws IOException
+  {
+    if (sslConfig != null) {
+      String nodeLocalConfig = sslConfig.getNodeLocalConfigPath();
+
+      if (Strings.isNotEmpty(nodeLocalConfig)) {
+        // all others should be empty
+        if (Strings.isNotEmpty(sslConfig.getKeyStorePath()) || Strings.isNotEmpty(sslConfig.getKeyStorePassword())
+            || Strings.isNotEmpty(sslConfig.getKeyStoreKeyPassword())) {
+          throw new IllegalArgumentException("Cannot specify both nodeLocalConfigPath and other parameters in " + sslConfig);
+        }
+        // pass thru: Stram will implement read the node local config file to implement SSL
+      } else {
+        // need to package and copy the keyStore file
+        String keystorePath = sslConfig.getKeyStorePath();
```

Comment:
I can't think of any validation on the passwords - they can be empty or anything. Only validation is on keystorePath that it is a valid file that can be read. But the next line (copyFromLocal) will throw a proper exception on failure so no need to check here unless you think we should check it to be a proper JKS file (which I don't agree with).